apollo-router-core/src/error.rs

```diff
@@ -111,13 +112,16 @@ impl FetchError {
 }
 
 /// Any error.
-#[derive(Error, Clone, Debug, Eq, PartialEq, Serialize, Deserialize, Default)]
+#[derive(Error, Clone, Debug, Eq, PartialEq, Serialize, Deserialize, Default, TypedBuilder)]
 #[error("{message}")]
 #[serde(rename_all = "camelCase")]
+#[builder(field_defaults(default, setter(strip_option)))]
```

Comment:
why is there a `strip_option` or `!strip_option` attribute everywhere? Could it be inserted only on `path`?